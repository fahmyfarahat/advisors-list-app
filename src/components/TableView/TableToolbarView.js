import React from "react";
import PropTypes from "prop-types";
import {
  Typography,
  Toolbar,
  Tooltip,
  IconButton,
  Drawer,
  FormControl,
  FormLabel,
  FormGroup,
  FormControlLabel,
  Checkbox,
  Button,
} from "@material-ui/core";
import FilterListIcon from "@material-ui/icons/FilterList";
import { useToolbarStyles } from "../../styles";

const TableToolbarView = ({ onChange, filters }) => {
  const classes = useToolbarStyles();
  const [isDrawerOpen, setIsDrawerOpen] = React.useState(false);

  const toggleDrawer = () => event => {
    if (event.type === "keydown" && (event.key === "Tab" || event.key === "Shift")) {
      return;
    }

    setIsDrawerOpen(!isDrawerOpen);
  };
  
  return (
    <Toolbar>
      <div className={classes.title}>
        <Typography variant="h6" id="tableTitle">
          Advisor list
        </Typography>
      </div>
      <div className={classes.spacer} />
      <div className={classes.actions}>
        <Tooltip onClick={toggleDrawer()} title="Filter list">
          <IconButton  aria-label="filter list">
            <FilterListIcon />
          </IconButton>
        </Tooltip>
        <Drawer anchor="right" open={isDrawerOpen} onClose={toggleDrawer()}>
          <div
            className={classes.list}
            role="presentation"
          >
            {Object.keys(filters).map(filterType => (
              <FormControl key={filterType} component="fieldset" className={classes.formControl}>
                 <FormLabel component="legend">{filterType}</FormLabel>
                 <FormGroup>
                   {Object.keys(filters[filterType]).map(option => (
                    <FormControlLabel
                      key={option}
                      control={<Checkbox checked={filters[filterType][option]} color="primary"
                      onChange={onChange(filterType, option)} value={option} />}
                      label={option}
                    />
                   ))}
                 </FormGroup>
              </FormControl>
            ))}
            <Button
              variant="contained"
              className={classes.button}
              onClick={toggleDrawer()}
            >close</Button>
          </div>
        </Drawer>
      </div>
    </Toolbar>
  );
};

TableToolbarView.propTypes = {
  onChange: PropTypes.func.isRequired, 
  filters: PropTypes.object.isRequired
};

export default TableToolbarView;