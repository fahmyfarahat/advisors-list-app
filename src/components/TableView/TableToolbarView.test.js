import React from "react";
import { shallow } from "enzyme";
import TableToolbarView from "./TableToolbarView";
import toJson from "enzyme-to-json";

describe("TableToolbarView - general ", () => {
  it("should match Snapshot", () => {
    const props = {
      onChange: () => {},
      filters: {
        status: {
          online: false,
          offline: false
        },
      }
    };
    const wrapper = shallow(<TableToolbarView {...props} />);
    expect(wrapper).toHaveLength(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  })
})
