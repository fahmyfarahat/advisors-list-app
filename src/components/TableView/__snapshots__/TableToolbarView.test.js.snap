// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableToolbarView - general  should match Snapshot 1`] = `
<WithStyles(ForwardRef(Toolbar))>
  <div
    className="makeStyles-title-3"
  >
    <WithStyles(ForwardRef(Typography))
      id="tableTitle"
      variant="h6"
    >
      Advisor list
    </WithStyles(ForwardRef(Typography))>
  </div>
  <div
    className="makeStyles-spacer-1"
  />
  <div
    className="makeStyles-actions-2"
  >
    <WithStyles(Tooltip)
      onClick={[Function]}
      title="Filter list"
    >
      <WithStyles(ForwardRef(IconButton))
        aria-label="filter list"
      >
        <FilterListIcon />
      </WithStyles(ForwardRef(IconButton))>
    </WithStyles(Tooltip)>
    <WithStyles(ForwardRef(Drawer))
      anchor="right"
      onClose={[Function]}
      open={false}
    >
      <div
        className="makeStyles-list-4"
        role="presentation"
      >
        <WithStyles(ForwardRef(FormControl))
          className="makeStyles-formControl-5"
          component="fieldset"
          key="status"
        >
          <WithStyles(ForwardRef(FormLabel))
            component="legend"
          >
            status
          </WithStyles(ForwardRef(FormLabel))>
          <WithStyles(ForwardRef(FormGroup))>
            <WithStyles(ForwardRef(FormControlLabel))
              control={
                <ForwardRef(WithStyles)
                  checked={false}
                  color="primary"
                  value="online"
                />
              }
              key="online"
              label="online"
            />
            <WithStyles(ForwardRef(FormControlLabel))
              control={
                <ForwardRef(WithStyles)
                  checked={false}
                  color="primary"
                  value="offline"
                />
              }
              key="offline"
              label="offline"
            />
          </WithStyles(ForwardRef(FormGroup))>
        </WithStyles(ForwardRef(FormControl))>
        <WithStyles(ForwardRef(Button))
          className="makeStyles-button-6"
          onClick={[Function]}
          variant="contained"
        >
          close
        </WithStyles(ForwardRef(Button))>
      </div>
    </WithStyles(ForwardRef(Drawer))>
  </div>
</WithStyles(ForwardRef(Toolbar))>
`;
